sphinx/ext/autosummary/generate.py

```diff
@@ -168,11 +176,26 @@ def generate_autosummary_docs(sources, output_dir=None, suffix='.rst',
             warn('[autosummary] failed to import %r: %s' % (name, e))
             continue
 
+        # https://docs.python.org/3/reference/import.html#packages
+        # ... any module that contains a __path__ attribute is
+        # considered a package ...
+        ispackage = hasattr(obj, '__path__')
+        if ispackage and recursive:
+            info(__('[autosummary] add modules/packages from %s') % repr(name))
+
         fn = os.path.join(path, name + suffix)
 
-        # skip it if it exists
+        # Skip it if it exists and not a package.
         if os.path.isfile(fn):
-            continue
+            if ispackage and recursive:
+                # Overwrite the file because modules/subpackages
+                # could have been added/removed from the package.
+                # Warning: this file could have been created by the user
+                # in which case it is lost.
+                warn('[autosummary] overwriting docs of package %s' % (repr(name)))
+                os.remove(fn)
```

Comment:
I will add an option for overwriting stub files in #6422.
So this special handling is not needed.

Reply:
Merged with master.